bundler/src/hook.rs

```diff
@@ -1,9 +1,12 @@
 use anyhow::Error;
 use swc_common::{FileName, Span};
-use swc_ecma_ast::Expr;
+use swc_ecma_ast::KeyValueProp;
 
 /// Note: As this is rarely used, it's recommended to pass it as a trait object.
 pub trait Hook: swc_common::sync::Sync + swc_common::sync::Send {
-    /// Return [None] to preserve `import.meta.url`
-    fn get_import_meta_url(&self, span: Span, file: &FileName) -> Result<Option<Expr>, Error>;
```

Comment:
I think the api should provide a way to determine if it's main.

Reply:
What do you think about this structure?
```rust
use anyhow::Error;
use swc_common::{FileName, Span};
use swc_ecma_ast::KeyValueProp;

#[non_exhaustive]
pub struct ModuleRecord {
  file_name: FileName,
  is_entry: bool,
}

pub trait Hook: swc_common::sync::Sync + swc_common::sync::Send {
    fn get_import_meta_props(
        &self,
        span: Span,
        module_record: &ModuleRecord,
    ) -> Result<Vec<KeyValueProp>, Error>;
}
```

Reply:
Done.

Reply:
Seems great! Thanks.